src/main/java/commands/DeleteCommand.java

```diff
@@ -0,0 +1,27 @@
+package commands;
+
+import storage.Storage;
+import tasks.Task;
+import tasks.TaskList;
+import ui.Ui;
+
+public class DeleteCommand extends Command {
+    private int taskIndex;
+
+    public DeleteCommand(int taskIndex) {
+        super();
+        this.taskIndex = taskIndex;
+    }
+
+    @Override
+    public void execute(TaskList taskList, Ui ui, Storage storage) {
+        Task deleted = taskList.deleteTask(taskIndex);
+        ui.showDelTaskMsg(deleted.toString(), taskList.getListSize());
```

Comment:
Is there a need for deleted.toString() as compared to simply passing deleted and printing out deleted in the ui?

Reply:
You're right, there's no need to call toString() here as it would be implicitly called upon.